Zend/zend_operators.c

```diff
@@ -333,7 +342,15 @@ static zend_never_inline zend_long ZEND_FASTCALL zendi_try_get_long(zval *op, bo
 					 * We use use saturating conversion to emulate strtol()'s
 					 * behaviour.
 					 */
-					 return zend_dval_to_lval_cap(dval);
+					lval = zend_dval_to_lval_cap(dval);
+					if (!zend_is_long_compatible(dval, lval)) {
+						zend_error(E_DEPRECATED, "Implicit conversion to int from non-compatible float-string %s",
```

Comment:
```suggestion
						zend_error(E_DEPRECATED, "Implicit conversion from non-compatible float-string \"%s\" to int",
```
As the order for the other error message changed, I'd change it here as well.